packages/xdl/src/Project.ts

```diff
@@ -381,20 +381,35 @@ export async function getLatestReleaseAsync(
     owner?: string;
   }
 ): Promise<Release | null> {
-  // TODO(ville): move request from multipart/form-data to JSON once supported by the endpoint.
-  let formData = new FormData();
-  formData.append('queryType', 'history');
-  formData.append('slug', await getSlugAsync(projectRoot));
-  if (options.owner) {
-    formData.append('owner', options.owner);
-  }
-  formData.append('version', '2');
-  formData.append('count', '1');
-  formData.append('releaseChannel', options.releaseChannel);
-  formData.append('platform', options.platform);
-  const { queryResult } = await Api.callMethodAsync('publishInfo', [], 'post', null, {
-    formData,
-  });
+  let result;
+  if (process.env.EXPO_LEGACY_API === 'true') {
+    // TODO(ville): move request from multipart/form-data to JSON once supported by the endpoint.
+    let formData = new FormData();
+    formData.append('queryType', 'history');
+    formData.append('slug', await getSlugAsync(projectRoot));
+    if (options.owner) {
+      formData.append('owner', options.owner);
+    }
+    formData.append('version', '2');
+    formData.append('count', '1');
+    formData.append('releaseChannel', options.releaseChannel);
+    formData.append('platform', options.platform);
+    result = await Api.callMethodAsync('publishInfo', [], 'post', null, {
+      formData,
+    });
+  } else {
+    const user = await UserManager.ensureLoggedInAsync();
+    const api = ApiV2.clientForUser(user);
+    result = await api.postAsync('publish/history', {
+      owner: options.owner,
+      slug: await getSlugAsync(projectRoot),
+      version: '2',
```

Comment:
I believe the `version` field should be removed from APIV2 code of this endpoint. It was used to denote two backwards incompatible versions of the APIV1 endpoint. For APIV2, we've never released any client code that uses the older variant, so this field is redundant.

We should also remove it from the backend here (since there are no clients calling the APIV2 history endpoint with version `1`:
https://github.com/expo/universe/blob/85f7202d0f1b295ed87d6a32b96186266032e3e9/server/www/src/apiv2/PublishRouter.ts#L499-L504

Reply:
Nice. Will do.

Reply:
Actually, regarding the backend code, there maybe some old clients that still call it. So as long as we have the deprecated apiv1 endpoints up, we should keep the code.

Reply:
And if the backend code stays the same, then we need to keep the version code here as well, otherwise it defaults to version===1 and deletes fields.

Reply:
The `version === 2` parameter was added to Expo CLI more than 2 years ago in https://github.com/expo/expo-cli/commit/3c4b73b98891b90c3a25e67bd16502fc93a387d1.

It's safe to say that any Expo CLI version older than that has been deprecated and is no longer supported. We should remove support for `version === 1`.

Reply:
OK, great.